plugins/module_utils/exceptions.py

```diff
@@ -29,6 +32,28 @@ def __init__(self, message=None, exception=None, **kwargs):
         # such as AnsibleLookupError, so can't easily consume this.
         self.kwargs = kwargs or {}
 
+    def is_boto3_error_code(self, code: Union[str, List[str]]) -> bool:
```

Comment:
one of the things I like about `botocore.is_boto3_error_code()` is that `except is_boto3_error_core(....):` will only catch the specific error rather than the 

```
try:
...
except SomeException as e:
  if not e.is_boto3_error_core(...):
     raise ...
  pass
```

Where it's way too easy to drop the wrong exceptions on the floor.

Reply:
I have updated with 2 similar functions dedicated `is_ansible_aws_error_message()` and `is_ansible_aws_error_code`